### PR TITLE
Eco 1307 sdk pull customization data from the backend endpoint

### PR DESCRIPTION
**Pull request by guido732**
## Motivation / Description
Implement the Appearance values coming from the branding endpoint into the components for the purchase flow

### CODE REVIEW

File: global.d.ts [hunk 1]
```diff
@@ -1,3 +1,5 @@
-export {}; // eslint-disable-line
+declare module "*.svelte" {
+  export { SvelteComponentDev as default } from "svelte/internal";
+}
```
Comment (guido732) [inline global.d.ts:3]: this is part of a fix that caused some warnings during imports about missing types

File: src/networking/responses/branding-response.ts [hunk 2]
```diff
@@ -1,6 +1,7 @@
 export type BrandingInfoResponse = {
   app_icon: string | null;
   app_icon_webp: string | null;
+  appearance: Record<string, string | boolean>;
```
Comment (guido732) [inline src/networking/responses/branding-response.ts:4]: haven't added the full type yet

**Review by tonidero — COMMENTED** [review]
Looking great! Just a question about the storage for the colors + possibly adding a type for the apparance to have more safety.

File: src/networking/responses/branding-response.ts (hunk 2 above)
Comment (guido732) [reply src/networking/responses/branding-response.ts:4]: I'll add them!

File: src/networking/responses/branding-response.ts [hunk 3]
```diff
@@ -1,6 +1,21 @@
...
+      color_accent: string;
+      color_error: string;
+      color_product_info_bg: string;
+      color_form_bg: string;
+      color_page_bg: string;
+      font: string;
+      shapes: "default" | "rectangle" | "rounded" | "pill";
+      show_product_description: boolean;
+    }
+  | Record<string, unknown>;
```
[the remark this answers is not in the document]
Comment (guido732) [reply src/networking/responses/branding-response.ts:13]: you're right, I modified this to undefined

[the remark this answers is not in the document]
Comment (tonidero) [reply src/networking/responses/branding-response.ts:13]: Hmm isn't it the same problem with undefined? We basically lose all type safety by allowing anything here... Is there a reason we should allow arbitrary data here?

File: src/networking/responses/branding-response.ts [hunk 4]
```diff
@@ -1,6 +1,21 @@
+export type BrandingAppearance =
+  | {
+      color_buttons_primary: string;
+      color_accent: string;
+      color_error: string;
+      color_product_info_bg: string;
+      color_form_bg: string;
+      color_page_bg: string;
+      font: string;
```
[the remark this answers is not in the document]
Comment (guido732) [reply src/networking/responses/branding-response.ts:9]: yes, it's either all or none

**Review by tonidero — APPROVED** [review]
LGTM! Thanks 🙌

**Pull request merged**